src/Transports.AspNetCore.NewtonsoftJson/GraphQLRequestDeserializer.cs

```diff
@@ -44,12 +44,28 @@ public Task<GraphQLRequestDeserializationResult> DeserializeFromJsonBodyAsync(Ht
                 switch (firstChar)
                 {
                     case '{':
-                        result.Single = ToGraphQLRequest(_serializer.Deserialize<InternalGraphQLRequest>(jsonReader));
+                        try
```

Comment:
I would not add try/catch inside each deserializer. Let's exception bubble up the stack and catch it in `GraphQLHttpMiddleware` so you don't need new `Exception` property.  Even further. Now I think that `IsSuccessful` property is redundant since we can just throw exception. In this case, API will become simpler - either good deserialized result or exception, no third option via `bool` flag.

Reply:
The was my first thoughts as well, but _serializer.Deserialize will use either `Newtonsoft.Json` or `System.Text.Json`. Which means we would either have to:
1. Catch both `Newtonsoft.Json.JsonException` and `System.Text.Json.JsonException` in middleware, which makes for a very leaky abstraction.
2. Catch `Exception` and assume it was a a json parsing exception :(
3. Make a wrapper exception, e.g. `GraphQLRequestDeserializerJsonException` to be thrown be both deserializers.

Reply:
I have implemented 3. using a wrapper exception.